src/app/services/errors/error-parser.service.ts

```diff
@@ -92,15 +92,86 @@ export class ErrorParserService {
     return null;
   }
 
+  private extractErrorDetails(error: ApiErrorDetails): ErrorDetails[] {
+    const details: ErrorDetails[] = [];
+
+    // Add error name if present
+    if (error.errname) {
+      details.push({ label: 'Error Name', value: error.errname });
+    }
+
+    // Add error code if present
+    if (error.error !== undefined && error.error !== null) {
+      details.push({ label: 'Error Code', value: error.error });
+    }
+
+    // Add reason if present
+    if (error.reason) {
+      details.push({ label: 'Reason', value: error.reason });
+    }
+
+    // Add error class from trace if present
+    if (error.trace?.class) {
+      details.push({ label: 'Error Class', value: error.trace.class });
+    }
+
+    // Add extra data if present
+    if (error.extra !== undefined && error.extra !== null) {
+      if (typeof error.extra === 'object' && !Array.isArray(error.extra)) {
+        Object.entries(error.extra).forEach(([key, value]) => {
+          details.push({
+            label: key.replace(/_/g, ' ').replace(/\b\w/g, (letter) => letter.toUpperCase()),
+            value: typeof value === 'object' ? JSON.stringify(value, null, 2) : String(value),
+          });
+        });
+      } else {
+        details.push({ label: 'Extra', value: JSON.stringify(error.extra, null, 2) });
+      }
+    }
+
+    // Add formatted trace if present (usually contains stack trace)
+    if (error.trace?.formatted) {
+      details.push({ label: traceDetailLabel, value: error.trace.formatted });
+    }
+
+    return details;
+  }
+
   private parseApiError(error: ApiErrorDetails): ErrorReport {
+    // Handle network connectivity errors with specific messages
+    if (error.errname === ApiErrorName.ConnectionReset || error.errname === ApiErrorName.TimedOut) {
+      return {
+        title: this.translate.instant('Network Error'),
+        message: this.translate.instant('Network connection was closed or timed out. Try again later.'),
+        icon: 'ix-cloud-off',
+        details: this.extractErrorDetails(error),
+      };
+    }
+    if (error.errname === ApiErrorName.NetworkUnreachable) {
+      return {
+        title: this.translate.instant('Network Error'),
+        message: this.translate.instant('Network resource is not reachable, verify your network settings and health.'),
+        hint: this.translate.instant('Double check that your nameservers and gateway are properly configured.'),
```

Comment:
@Mrt134 @tonyriv3 do you think we can get a review on the text for this method, please?